bin/build_cmd.ml

```diff
@@ -231,6 +231,17 @@ let fmt =
     in
     let common, config = Common.init builder in
     let request (setup : Import.Main.build_system) =
+      let open Action_builder.O in
+      let* () =
+        if Lazy.force Lock_dev_tool.is_enabled
```

Comment:
You can keep this around if you'd like to experiment with this, but I'd like to re-iterate that enabling this behavior being enabled is unlikely.  The job of `dune build` is to execute build rules. So I think you'd need a way to turn this into a build rule before this could be enabled.

Reply:
Just to clarify, are you saying that it's not acceptable to make the current behaviour the default, but it's fine to upstream it disabled by default? Is the problem that with this feature enabled, running `dune build` runs the solver in addition to executing build rules?

Reply:
> Just to clarify, are you saying that it's not acceptable to make the current behaviour the default, but it's fine to upstream it disabled by default?

That's correct.

>  Is the problem that with this feature enabled, running dune build runs the solver in addition to executing build rules?

I'd rephrase it this way: running the solver does not equate building a target in our world. `$ dune build` is for building targets

Reply:
Currently this feature will only run the solver if the user types `dune fmt`, not `dune build`. Is that still a problem?

Reply:
Nope, that's not a problem. I think the comment below says the intention is to move it to `dune build` though.

Reply:
Ok I updated the comment to clarify that we don't intend to move this logic to `dune build`.